app/controllers/api/v1/journeys_controller.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class JourneysController < ApiController
+      before_action :validate_params, only: %i[create update]
+      after_action :create_event, only: %i[create update]
+
+      PERMITTED_NEW_JOURNEY_PARAMS = [
+          :type,
+          attributes: [:timestamp, :billable, vehicle: {}],
+          relationships: [from_location: {}, to_location: {}],
+      ].freeze
+
+      PERMITTED_UPDATE_JOURNEY_PARAMS = [
+          :type,
+          attributes: [:timestamp, :billable, vehicle: {}],
+      ].freeze
+
+      def index
+        paginate journeys
+      end
+
+      def show
+        render json: journey, status: :ok
+      end
+
+      def create
+        authorize!(:create, journey)
```

Comment:
I'm not entirely clear why this extra `authorize!` call is necessary? The `journey` method already chains `move` method, which in turn scopes to moves accessible to the supplier. So if a supplier attempts to create a journey relating to a move they don't have access to then I would expect the authorisation to fail within the `move` method. So presumably that all works even without this line?

Reply:
@smoothcontract good question: its to catch a user trying to create a journey assigned to a different supplier. Both moves and journeys have to be authorised; the move is authorised by its scope; but the journey is new and not yet persisted to the database - hence the authorisation check.

Reply:
Cool, that makes sense - thanks for the explanation 👍